lib/fluent/plugin/out_s3.rb

```diff
@@ -76,19 +78,15 @@ def start
       if @aws_key_id && @aws_sec_key
         options[:access_key_id] = @aws_key_id
         options[:secret_access_key] = @aws_sec_key
-      elsif ENV.key? "AWS_ACCESS_KEY_ID"
-        options[:credential_provider] = AWS::Core::CredentialProviders::ENVProvider.new('AWS')
-      else
-        options[:credential_provider] = AWS::Core::CredentialProviders::EC2Provider.new({:retries => @aws_iam_retries})
       end
       options[:region] = @s3_region if @s3_region
       options[:s3_endpoint] = @s3_endpoint if @s3_endpoint
       options[:proxy_uri] = @proxy_uri if @proxy_uri
-      options[:use_ssl] = @use_ssl
```

Comment:
Apparently so. `:endpoint` seems to be the new option.
https://github.com/aws/aws-sdk-ruby/issues/746

From a quick grep search, `proxy_uri` appears to be gone as well.